custom_components/frank_energie/const.py

```diff
@@ -113,24 +113,28 @@ class FrankEnergieEntityDescription(SensorEntityDescription):
         name="Lowest gas price today",
         native_unit_of_measurement=f"{CURRENCY_EURO}/{VOLUME_CUBIC_METERS}",
         value_fn=lambda data: min(data['today_gas']),
+        attr_fn=lambda data: {ATTR_HOUR: data['today_gas'].index(min(data['today_gas']))},
```

Comment:
Dit index-trucje werkt helaas niet altijd helemaal lekker. Mijn gasprijs zou nu het hoogst zijn om 30:00. Daarnaast zou het het netst zijn om een tijdobject te retourneren in plaats van een string of int.

<img width="896" alt="Schermafbeelding 2022-10-09 om 12 29 30" src="https://user-images.githubusercontent.com/6934851/194751766-230360f8-e4eb-4ede-a346-41ae9db69bef.png">

Reply:
> Dit index-trucje werkt helaas niet altijd helemaal lekker. Mijn gasprijs zou nu het hoogst zijn om 30:00. Daarnaast zou het het netst zijn om een tijdobject te retourneren in plaats van een string of int.
> 
> <img alt="Schermafbeelding 2022-10-09 om 12 29 30" width="896" src="https://user-images.githubusercontent.com/6934851/194751766-230360f8-e4eb-4ede-a346-41ae9db69bef.png">

Als jij 30 uur laat terugkomen in deze functie dan klopt de naam "today" ook niet maar is het "today & stukje morgen"
Om die reden zorg ik ervoor dat er altijd maar 24 uur terugkomen met deze functie (voor electra)
Voor gas is het anders omdat er eigenlijk geen hoogste en laagste prijs is omdat het maar  1 keer om 6:00 uur veranderd
Met gas ben ik bijna klaar...alleen nog even testen of alles klopt op de juiste uren.
today_hours wil ik terug laten komen als 2e waarde bij return. Ik ben er alleen nog niet uit hoe dat werkt.
```
        i=0
        for hour in hourprices:
            if i < 24:
                yesterday_prices.append(
                    (hour['marketPrice'] + hour['marketPriceTax'] + hour['sourcingMarkupPrice'] + hour['energyTaxPrice'])
                )
            if 23 < i < 48:
                today_prices.append(
                    (hour['marketPrice'] + hour['marketPriceTax'] + hour['sourcingMarkupPrice'] + hour['energyTaxPrice'])
                )
                today_hours.append(
                    (hour['from'])
                )
            if 47 < i < 72:
                tomorrow_prices.append(
                    (hour['marketPrice'] + hour['marketPriceTax'] + hour['sourcingMarkupPrice'] + hour['energyTaxPrice'])
                )
            i=i+1
        if 3 < datetime.now().hour < 24:
            return today_prices
        if -1 < datetime.now().hour < 3:
            if tomorrow_prices:
                return tomorrow_prices
        return today_prices
```